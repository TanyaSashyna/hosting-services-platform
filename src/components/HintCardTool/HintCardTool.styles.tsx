import styled from "styled-components";
import {
  ButtonGroup as _ButtonGroup,
  paddings,
  breakpoints,
} from "@project-name-frontend/ui-package";

export const ButtonGroup = styled(_ButtonGroup)`
  .hint-btn {
    @media (max-width: ${breakpoints.phoneUpperBoundary}) {
      .button__internal {
        padding-left: 0;
        padding-right: 0;
      }
    }
  }
`;

export const TextWrap = styled("div")`
  margin-bottom: ${paddings.default};
`;
