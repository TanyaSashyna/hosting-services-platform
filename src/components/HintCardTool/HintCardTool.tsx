import React, { FC, useState } from "react";
import { HintCard, Button, Link } from "@project-name-frontend/ui-package";

import { ButtonGroup, TextWrap } from "./HintCardTool.styles";

export const HintCardTool: FC = () => {
  const [open, setOpen] = useState<boolean>(false);

  return open ? (
    <HintCard
      header="What is Tool?"
      body={[
        {
          key: 1,
          content: (
            <>
              <div>
                Tool is a secure, scalable system of servers that store (cache)
                website content closer to end users. Our collaboration with{" "}
                <Link href="https://..." newTab={true}>
                  Fastly
                </Link>{" "}
                lets you add Tool straight from your control panel to your chosen
                domain.
              </div>
              <div className="hintCard__head">How does it work?</div>
              <TextWrap>
                Tool acts as a temporary house for your websiteâ€™s content. When a
                customer visits your domain, Tool:
              </TextWrap>
            </>
          ),
        },
      ]}
      list={[
        {
          key: 1,
          content: "Compresses your web files to improve load times.",
        },
        {
          key: 2,
          content:
            "Locates the nearest available server, delivering cached content to your end user, directly at the edge.",
        },
      ]}
      listClassName="hintCard__body__listOne"
      onClose={(): void => setOpen(false)}
    />
  ) : (
    <ButtonGroup layout="right">
      <Button
        buttonType="borderless"
        onClick={(): void => setOpen(true)}
        rightIcon="keyboard_arrow_down"
        className="hint-btn"
      >
        About Tool
      </Button>
    </ButtonGroup>
  );
};

HintCardTool.displayName = "HintCardTool";
