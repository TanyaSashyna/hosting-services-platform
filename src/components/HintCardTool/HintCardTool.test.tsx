import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { HintCardTool } from "./HintCardTool";

afterEach(cleanup);

describe("HintCardList", () => {
  it("renders HintCardList", () => {
    const { container, getByText } = render(<HintCardTool />);
    expect(container).toBeDefined();
    expect(getByText("About Tool")).toBeDefined();
  });

  it("renders when HintCardList is open", () => {
    const { getByText } = render(<HintCardTool />);

    const btn = getByText("About Tool");
    fireEvent.click(btn);

    expect(getByText("What is Tool?")).toBeDefined();
  });

  it("renders when HintCardList is open", () => {
    const { getByText } = render(<HintCardTool />);

    const btn = getByText("About Tool");
    fireEvent.click(btn);

    const closeBtn = getByText("close");
    fireEvent.click(closeBtn);

    expect(() => getByText("What is Tool?")).toThrow(
      "Unable to find an element"
    );
  });

  it("matches snapshot", () => {
    const { container, asFragment } = render(<HintCardTool />);
    expect(container).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
