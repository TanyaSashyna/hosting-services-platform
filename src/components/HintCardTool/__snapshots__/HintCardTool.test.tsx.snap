// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintCardList matches snapshot 1`] = `
<DocumentFragment>
   
  @media (max-width:768px) {
  .c0 .hint-btn .button__internal {
    padding-left: 0;
    padding-right: 0;
  }
}

<div
    class="buttonGroup _buttonGroup--layout-center c0 buttonGroup--mobile"
  >
    <button
      class="button hint-btn button--size-default button--type-borderless button--text"
      type="button"
    >
      <div
        class="button__internal"
      >
        <div
          class="button__children"
        >
          About Tool
        </div>
        <div
          class="button__icon"
        >
          <i
            class="icon"
          >
            keyboard_arrow_down
          </i>
        </div>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
